File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n = nums.length;
        int[] prefix = new int[n + 1];
        
        for(int i = 1; i <=n; i++) prefix[i] = prefix[i - 1] + nums[i - 1];
        int res = Integer.MAX_VALUE, l = 0;
        for(int i = 1; i <= n; i++) {
            while(prefix[i] - prefix[l] >= target) {
                res = Math.min(res, i - l );
                l++;
            }
        }
        return res == Integer.MAX_VALUE ? 0 : res;
    }
}

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        l, n, res, csum = 0, len(nums), float('inf'), 0
        for i in range(n):
            csum += nums[i]
            while csum >= target:
                res = min(res, i - l + 1)
                csum -= nums[l]
                l += 1
        return 0 if res == float('inf') else res
                